src/main/java/pl/zimi/zadanka/Zad5.java

```diff
@@ -0,0 +1,62 @@
+package pl.zimi.zadanka;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * source:
+ * https://leetcode.com/problems/climbing-stairs/
+ */
+
+public class Zad5 {
+
+    // rozwiązanie poprawne ale wolne
+    public static int climbStairsTreeStyle(int key, int currentValue, int carry) {
+        if (currentValue == key) {
+            return ++carry;
+        }
+        if (currentValue > key) {
+            return 0;
+        } else {
+            return climbStairsTreeStyle(key, currentValue + 1, carry)
+                    + climbStairsTreeStyle(key, currentValue + 2, carry);
+        }
+    }
+
+    // rozwiązanie poprawne ale za wolne
+    public static int climbStairs(int key) {
+        if (key > 0) {
+            if (key == 1) {
+                return 1;
+            }
+            if (key == 2) {
+                return 2;
+            }
```

Comment:
kod jest poprawny ale można go uprościć
zauważ że kod dla key = 0 ma zwrócić 0
dla key = 1 -> 1
dla key = 2 -> 2
dla większych wartości zachowanie się zmienia

ponadto cechą dobrego kodu jest reakcja na inne dane...
co powinno się wydarzyć gdy za key dostaniesz ujemną wartość

Reply:
zapropowanieł wywołanie wyjątku :
throw new IllegalArgumentException("input variable StairNumber cannot be less than 0");